src/interface/comm.rs

```diff
@@ -98,15 +98,15 @@ pub struct SockaddrV6 {
 
 #[derive(Debug)]
 pub struct Socket {
-    raw_sys_fd: i32//make private right after
+    pub refcnt: i32,
+    raw_sys_fd: i32
 }
 
 impl Socket {
     pub fn new(domain: i32, socktype: i32, protocol: i32) -> Socket {
         let fd = unsafe {libc::socket(domain, socktype, protocol)};
         if fd < 0 {panic!("Socket creation failed when it should never fail");}
-        unsafe {libc::fcntl(fd, libc::F_SETFL, libc::O_NONBLOCK);}
-        Socket {raw_sys_fd: fd}
+        Self {refcnt: 1, raw_sys_fd: fd}
```

Comment:
Is this change necessary? I think that this breaks testing on MacOS.

Reply:
This shouldn't because as per the meeting last week we need not do the nonblocking behavior at all, so not doing it in either fcntl or in the send/recv calls would be fine for mac.